File: particle-base/ios/ParticleBasePlugin.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ParticleBasePlugin, NSObject)

RCT_EXTERN_METHOD(initialize: (NSString* _Nonnull)json)

RCT_EXTERN_METHOD(setChainInfo: (NSString* _Nonnull)json callback:(RCTResponseSenderBlock)callback)

RCT_EXTERN_METHOD(getChainInfo: (RCTResponseSenderBlock)callback)

RCT_EXTERN_METHOD(setInterfaceStyle:(NSString* _Nonnull)json)

RCT_EXTERN_METHOD(setLanguage:(NSString* _Nonnull)json)

RCT_EXTERN_METHOD(getLanguage: (RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(setAppearance:(NSString* _Nonnull)json)

RCT_EXTERN_METHOD(setFiatCoin:(NSString* _Nonnull)json)

RCT_EXTERN_METHOD(setSecurityAccountConfig:(NSString* _Nonnull)json)

RCT_EXTERN_METHOD(getSecurityAccount: (RCTResponseSenderBlock)callback)

- (dispatch_queue_t)methodQueue {
  return dispatch_get_main_queue();
}

@end

File: particle-base/ios/ParticleBasePlugin.swift
//
//  ParticleBasePlugin.swift
//  ParticleBasePlugin
//
//  Created by link on 2022/9/28.
//

import Base58_swift
import Foundation
import ParticleNetworkBase
import RxSwift
import SwiftyJSON

@objc(ParticleBasePlugin)
class ParticleBasePlugin: NSObject {
    let bag = DisposeBag()
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc
    public func initialize(_ json: String) {
        ShareBase.shared.initialize(json)
    }
    
    @objc
    public func setChainInfo(_ json: String, callback: @escaping RCTResponseSenderBlock) {
        let value = ShareBase.shared.setChainInfo(json)
        callback([value])
    }

    @objc
    public func getChainInfo(_ callback: @escaping RCTResponseSenderBlock) {
        let value = ShareBase.shared.getChainInfo()
        callback([value])
    }
    
    @objc
    public func setAppearance(_ json: String) {
        ShareBase.shared.setAppearance(json)
    }
    
    @objc
    public func setFiatCoin(_ json: String) {
        ShareBase.shared.setFiatCoin(json)
    }

    @objc
    public func setLanguage(_ json: String) {
        ShareBase.shared.setLanguage(json)
    }
    
    @objc
    public func getLanguage(_ resolve: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        let value = ShareBase.shared.getLanguage()
        resolve(value)
    }
    
    @objc
    public func setSecurityAccountConfig(_ json: String) {
        ShareBase.shared.setSecurityAccountConfig(json)
    }
    
    @objc func setThemeColor(_ json: String) {
        ShareBase.shared.setThemeColor(json)
    }
    
    @objc func setCustomUIConfigJsonString(_ json: String) {
        ShareBase.shared.setCustomUIConfigJsonString(json)
    }
    
    @objc func setUnsupportCountries(_ json: String) {
        ShareBase.shared.setUnsupportCountries(json)
    }
}
